projects/packages/waf/src/brute-force-protection/class-shared-functions.php

```diff
@@ -92,6 +101,29 @@ public static function get_global_whitelist() {
 		return $whitelist;
 	}
 
+	/**
+	 * Convert a string into an IP Address object.
+	 *
+	 * @param string $ip_address The IP Address to convert.
+	 * @return object An IP Address object.
+	 */
+	private static function create_ip_object( $ip_address ) {
+		$range = false;
+		if ( strpos( $ip_address, '-' ) ) {
+			$ip_address = explode( '-', $ip_address );
+			$range      = true;
+		}
+		$new_item        = new \stdClass();
+		$new_item->range = $range;
+		if ( ! empty( $range ) ) {
```

Comment:
Useless use of `empty()`, you know `$range` exists and is set because you just did it above.
```suggestion
		if ( $range ) {
```
I see a bunch more of this sort of thing further on in the PR too. Ideally you should only use `! empty( $x )` when you need to do `isset( $x ) && $x` (and `empty( $x )` when you need to do `! isset( $x ) || ! $x`). If you don't need the `isset()` to avoid PHP warnings about unset variables or whatever, you don't need `empty()` either.

Reply:
Thanks Brad! 

I've merged both of your suggestions. The other implementations of `empty()` are using the function to check if values are set and also not empty strings or arrays, so I've left those alone.